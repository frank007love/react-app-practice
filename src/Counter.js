import React, { useContext } from "react"
import { CountCtx } from "./Example"

function Counter() {
    const [count, setCounter] = useContext(CountCtx)
    return (
        <>
            <div aria-label="counter-val">Counter is {count}</div>
            <button onClick={() => {setCounter(count + 5);
            console.log('shit:', count);    
        }}>child button +5</button>
        </>
    )
}

export default Counter