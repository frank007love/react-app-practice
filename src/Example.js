import React, { useState, createContext } from "react"
import Counter from "./Counter"

export const CountCtx = createContext([0, () => {}])

function Example() {
    const [count, setCount] = useState(0);

    return (
        <div>
            <CountCtx.Provider value={[count, setCount]}>
                <Counter />
                <button onClick={() => setCount(count + 1)}>parent button +1</button>
            </CountCtx.Provider>
        </div>
    )
}

export default Example