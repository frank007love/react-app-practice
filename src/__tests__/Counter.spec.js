import Counter from '../Counter';
import { CountCtx } from "../Example";
import {render, fireEvent, cleanup} from 'react-testing-library';
import React, { useState } from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';

function CounterComp(){
    const [ count, setCount ] = useState(0);
    return <CountCtx.Provider value={[count, setCount]}><Counter/></CountCtx.Provider>;
}


describe('<Counter>',()=>{
    afterEach(cleanup);

    it('Counter snapshot', () => {
        const tree = renderer.create(<CounterComp/>).toJSON()
        expect(tree).toMatchSnapshot();
    })

    it('Counter should be added after clicking the button', ()=>{
        //const {queryByLabelText, getByLabelText} = render(<CounterComp/>);
        
        
        //const counter = shallow(<CounterComp/>);
        //expect(counter.find('div').text()).toBe('Counter is 0');
        
        //counter.find('button').simulate('click');
        //expect(counter.find('div').text()).toBe('Counter is 5');
        // console.log(form.find('Button'));
        // expect(form.find('Button').prop('disabled')).toBe(true);
        //expect(form.find('button').props()["disabled"]).toBe(true);
    });
});