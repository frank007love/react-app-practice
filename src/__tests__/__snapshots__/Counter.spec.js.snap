// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter> Counter snapshot 1`] = `
Array [
  <div>
    Counter is 
    0
  </div>,
  <button
    onClick={[Function]}
  >
    child button +5
  </button>,
]
`;
